Sources/SlidingContainerViewController.swift

```diff
@@ -0,0 +1,400 @@
+        upperController.viewController.didMove(toParentViewController: self)
+
+        let panGesture = UIPanGestureRecognizer(
+            target: self,
+            action: #selector(self.panDragView(sender:))
+        )
+        panGesture.delegate = self
+        panGesture.delaysTouchesBegan = false
+        panGesture.delaysTouchesEnded = false
+        panGesture.cancelsTouchesInView = false
+        navView?.addGestureRecognizer(panGesture)
+    }
+
+    open override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+        if !initialPositionSetUp {
+            initialPositionSetUp = true
+            drawViews()
+        }
+        lowerController.viewController.view.frame = frameForLowerController()
+        upperController.viewController.view.frame = frameForUpperController()
+    }
+
+    func getMovementDifference() -> CGFloat {
+        guard let navView = navView else { return 0 }
+        let diff = navView.frame.height - presenter.upperViewMin(for: self) - presenter.lowerViewMin(for: self) - (dragView?.frame.height ?? 42)
+        return diff
+    }
+
+    // MARK: Frame Management
+
+    func drawViews() {
+        guard let navView = navView else { return }
+        let navHeight = navView.frame.height
+        let navWidth = navView.frame.width
+        let dragHeight = (dragView?.frame.height ?? 42)
+        let middle = CGRect(
+            x: 0,
+            y: navHeight - presenter.lowerViewMin(for: self) - dragHeight,
+            width: navWidth,
+            height: dragHeight
+        )
+        let upper = CGRect(
+            x: 0,
+            y: 0,
+            width: navWidth,
+            height: navHeight - presenter.lowerViewMin(for: self) - dragHeight
+        )
+        let lower = CGRect(
+            x: 0,
+            y: navHeight - presenter.lowerViewMin(for: self),
+            width: navWidth,
+            height: presenter.lowerViewMin(for: self)
+        )
+
+        upperView.frame = upper
+        dragView?.frame = middle
+        lowerView.frame = lower
+    }
+
+    func frameForLowerController() -> CGRect {
+        let lower = CGRect(
+            x: 0,
+            y: 0,
+            width: lowerView.frame.width,
+            height: lowerView.frame.height
+        )
+        return lower
+    }
+
+    func frameForUpperController() -> CGRect {
+        let upper = CGRect(
+            x: 0,
+            y: 0,
+            width: upperView.frame.width,
+            height: upperView.frame.height
+        )
+        return upper
+    }
+
+    func updateViews(with translation: CGPoint, forState state: UIGestureRecognizerState) {
+        var f0 = dragView?.frame ?? .zero
+        var f1 = upperView.frame
+        var f2 = lowerView.frame
+
+        switch presenter.movementType {
+        case .hideUpperPushLower:
+            if state == .ended {
+                if
+                    let navHeight = navView?.frame.height,
```

Comment:
please move all the condition of this `if` to an auxiliar function and reuse it from everywhere